account_invoice_auto_send_mail/models/account_invoice.py

```diff
@@ -0,0 +1,88 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import api, fields, models
+from datetime import datetime
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    date_invoice_send_mail = fields.Datetime(
+        string='Date invoice send mail'
+    )
+
+    @api.one
+    def account_invoice_auto_send_mail_item_real(self, mail_template_id, author_id):
+        _logger.info('Operations account_invoice_auto_send_mail_item_real invoice ' + str(self.id))
+
+        mail_compose_message_vals = {
+            'author_id': self.user_id.partner_id.id,
+            'record_name': self.number,
+        }
+        # Author_id (journal_id)
+        if author_id.id > 0:
```

Comment:
```suggestion
        if author:
```
I would prefer to name the variable, with `_id`, `_ids`, it it keeps number. And without `_id`, `_ids` for recordsets.